File: app/src/main/java/com/example/agroapp/ui/main/activity/MainActivity.java
package com.example.agroapp.ui.main.activity;

import android.app.ActionBar;
import android.content.pm.ActivityInfo;
import android.os.Bundle;

import androidx.core.content.ContextCompat;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;
import androidx.lifecycle.ViewModelProviders;

import com.example.agroapp.BR;
import com.example.agroapp.R;
import com.example.agroapp.app.AppViewModelFactory;
import com.example.agroapp.databinding.ActivityMainBinding;
import com.example.agroapp.ui.main.fragment.DiscoverFragment;
import com.example.agroapp.ui.main.fragment.HomepageFragment;
import com.example.agroapp.ui.main.fragment.NavigationFragment;
import com.example.agroapp.ui.main.fragment.ProjectFragment;

import java.util.ArrayList;
import java.util.List;

import me.goldze.mvvmhabit.base.BaseActivity;
import me.goldze.mvvmhabit.base.BaseViewModel;
import me.majiajie.pagerbottomtabstrip.NavigationController;
import me.majiajie.pagerbottomtabstrip.listener.OnTabItemSelectedListener;

/**
 * author : jiangxue
 * date : 2023/5/16 16:06
 * description :首页
 */
public class MainActivity extends BaseActivity<ActivityMainBinding, BaseViewModel> {
    private List<Fragment> mFragments;

    @Override
    public int initContentView(Bundle savedInstanceState) {
        return R.layout.activity_main;
    }

    @Override
    public int initVariableId() {
        return BR.viewModel;
    }
    @Override
    public void initParam() {
        super.initParam();
        setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
    }
    @Override
    public void initData() {
        //初始化Fragment
        initFragment();
        //initToolbar();
        //初始化底部Button
        initBottomTab();
    }



    private void initFragment() {
        mFragments = new ArrayList<>();
        mFragments.add(new HomepageFragment());
        mFragments.add(new DiscoverFragment());
        mFragments.add(new HomepageFragment());
        mFragments.add(new NavigationFragment());
        mFragments.add(new ProjectFragment());

        //默认选中第一个
        commitAllowingStateLoss(0);
    }
    private void initToolbar() {

        //setSupportActionBar(binding.iToolbar.tbToolbar);
        ActionBar actionBar = getActionBar();
        actionBar.setHomeAsUpIndicator(R.drawable.ic_menu);
        actionBar.setDisplayHomeAsUpEnabled(true);
        //actionBar.setTitle(R.string.app_name);
        actionBar.setTitle("测试APP");
    }
    private void initBottomTab() {
        NavigationController navigationController = binding.pagerBottomTab.material()
                .addItem(R.drawable.ic_homepage, "首页")
                .addItem(R.drawable.ic_system, "体系")
                .addItem(R.drawable.ic_we_chat, "公众号")
                .addItem(R.drawable.ic_navigation, "导航")
                .addItem(R.drawable.ic_project, "项目")
                .setDefaultColor(ContextCompat.getColor(this, R.color.textColorVice))
                .build();
        //底部按钮的点击事件监听
        navigationController.addTabItemSelectedListener(new OnTabItemSelectedListener() {
            @Override
            public void onSelected(int index, int old) {
                commitAllowingStateLoss(index);
               // binding.iToolbar.tbToolbar.setTitle();
            }

            @Override
            public void onRepeat(int index) {
            }
        });
    }

    private void commitAllowingStateLoss(int position) {
        hideAllFragment();
        FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
        Fragment currentFragment = getSupportFragmentManager().findFragmentByTag(position + "");
        if (currentFragment != null) {
            transaction.show(currentFragment);
        } else {
            currentFragment = mFragments.get(position);
            transaction.add(R.id.frameLayout, currentFragment, position + "");
        }
        transaction.commitAllowingStateLoss();
    }

    //隐藏所有Fragment
    private void hideAllFragment() {
        FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
        for (int i = 0; i < mFragments.size(); i++) {
            Fragment currentFragment = getSupportFragmentManager().findFragmentByTag(i + "");
            if (currentFragment != null) {
                transaction.hide(currentFragment);
            }
        }
        transaction.commitAllowingStateLoss();
    }

}

File: kotlin/src/main/java/com/example/kotlin/ui/activity/main/MainActivity.kt
package com.example.kotlin.ui

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.Fragment
import com.example.kotlin.R
import com.example.kotlin.databinding.ActivityMainBinding
import com.example.kotlin.ui.fragment.home.HomepageFragment
import me.majiajie.pagerbottomtabstrip.NavigationController
import java.util.*

//public class MainActivity extends BaseActivity<ActivityMainBinding, BaseViewModel> {
class MainActivity : AppCompatActivity() {
    protected final var TAG : String = "MainActivity";
    private var mFragments: ArrayList<Fragment>? = null
   //在使用lateinit 定义的变量前 一定会给他一个实例 保证他不会是空对象 而 by lazy 则是在第一次使用时 初始化对象
    private val binding by lazy { DataBindingUtil.setContentView<ActivityMainBinding>(this, R.layout.activity_main) }
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        //初始化Fragment
        initFragment()
        //initToolbar();
        //初始化底部Button
        initBottomTab()
    }




    private fun initFragment() {
        mFragments = ArrayList()
        mFragments!!.add(HomepageFragment())
        mFragments!!.add(HomepageFragment())
        mFragments!!.add(HomepageFragment())
        mFragments!!.add(HomepageFragment())

        //默认选中第一个
        commitAllowingStateLoss(0)
    }

    private fun initToolbar() {

        //setSupportActionBar(binding.iToolbar.tbToolbar);
        val actionBar = actionBar

        //actionBar.setHomeAsUpIndicator(R.drawable.ic_menu)
        actionBar!!.setDisplayHomeAsUpEnabled(true)
        //actionBar.setTitle(R.string.app_name);
        actionBar.title = "测试APP"
    }

    private fun initBottomTab() {
        val navigationController: NavigationController = binding!!.pagerBottomTab.material()
                .addItem(R.drawable.ic_homepage, "首页")
                .addItem(R.drawable.ic_system, "体系")
                .addItem(R.drawable.ic_we_chat, "公众号")
                .addItem(R.drawable.ic_navigation, "导航")
                .addItem(R.drawable.ic_project, "项目")
                .build()
    }

    private fun commitAllowingStateLoss(position: Int) {
        hideAllFragment()
        val transaction = supportFragmentManager.beginTransaction()
        var currentFragment = supportFragmentManager.findFragmentByTag(position.toString() + "")
        if (currentFragment != null) {
            transaction.show(currentFragment)
        } else {
            currentFragment = mFragments!![position]
            transaction.add(R.id.frameLayout, currentFragment, position.toString() + "")
        }
        transaction.commitAllowingStateLoss()
    }

    //隐藏所有Fragment
    private fun hideAllFragment() {
        val transaction = supportFragmentManager.beginTransaction()

//personList.indices，可以得到所得对象的下标值
        for (i in mFragments!!.indices) {
            val currentFragment = supportFragmentManager.findFragmentByTag(i.toString() + "")
            if (currentFragment != null) {
                transaction.hide(currentFragment)
            }
        }
        transaction.commitAllowingStateLoss()
    }
}


